subsys/bluetooth/audio/shell/bap_broadcast_assistant.c

```diff
@@ -49,6 +54,16 @@ static struct bt_auto_scan {
 	.broadcast_id = INVALID_BROADCAST_ID,
 };
 
+static struct bt_broadcast_name_scan {
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	char broadcast_name[MAX_BROADCASTNAME_LEN + 1];
```

Comment:
Why not just add char broadcast_name[..] to struct bt_auto_scan?

Reply:
In my opinion, it is cleaner to keep them separated. But it's just a matter of taste!

Reply:
I would probably keep them together. Maybe `union` the name and ID? 

The reason why is that we do not want both to be active at the same time, right? So effectively they should share the same "state". 

Not going to block this PR based on that, as this would also work.

Reply:
Using `union` is a good idea, I'll update the next commit based on it.